pkg/skaffold/docker/parse.go

```diff
@@ -75,6 +76,11 @@ func readCopyCmdsFromDockerfile(onlyLastImage bool, absDockerfilePath, workspace
 		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
 	}
 
+	// instructions.Parse will check for malformed Dockerfile
+	if _, _, err := instructions.Parse(res.AST); err != nil {
+		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
+	}
```

Comment:
Should this happen before we call `parser.Parse()`? Since we basically want to validate the dockerfile with `instructions.Parse()` before then putting the data from the Dockerfile into `res`

Also should the error message be different to give a little more context? Maybe "validating dockerfile" or something similar?

Reply:
> Should this happen before we call `parser.Parse()`? Since we basically want to validate the dockerfile with `instructions.Parse()` before then putting the data from the Dockerfile into `res`

Not sure what you mean, `parser.Parse()` output is a required input to `instructions.Parse()`.

> Also should the error message be different to give a little more context? Maybe "validating dockerfile" or something similar?

`instructions.Parse` itself returns a very detailed error message so it's safe to not add any more text here.

Reply:
Sorry, I misread, my bad. 

Sweet, thanks for clarifying the error message :)
I think this looks good then